### ISSUE TITLE
Compilation error in ScalaGenerator generated code when table has column "CLASS"

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10877]
When using `ScalaGenerator` to generate records for this table:

```sql
CREATE TABLE override_conflicts (
  class int
);
```

The generated output contains a compilation error:

```scala
  /**
   * Getter for <code>OVERRIDE_CONFLICTS.CLASS</code>.
   */
  def getClass_: Integer = get(0).asInstanceOf[Integer]
```

The error being:

```
10:50:38,800 [ERROR] [Error] C:\Users\lukas\jOOQ\3.15\workspace\jOOQ-pro\jOOQ-test-codegen-scala\src\test\scala\org\jooq\codegen\test\oracle\db\tables\records\TReservedRecord.scala:64: '=' expected but identifier found.
10:50:38,810 [ERROR] [Error] C:\Users\lukas\jOOQ\3.15\workspace\jOOQ-pro\jOOQ-test-codegen-scala\src\test\scala\org\jooq\codegen\test\oracle\db\tables\records\TReservedRecord.scala:69: illegal start of simple expression
```

Adding the parentheses seems to work:

```scala
  /**
   * Getter for <code>OVERRIDE_CONFLICTS.CLASS</code>.
   */
  def getClass_(): Integer = get(0).asInstanceOf[Integer]
```

Rather than investigating the exception in this particular case, we'll just re-add the parentheses everywhere.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.2 (#10878)

**Comment by joroKr21** [comment]
You just need a space here: `def getClass_ : Integer = get(0).asInstanceOf[Integer]`. This is because in Scala you can have an identifier that starts alphanumeric, underscore and then symbolic, e.g. `List(4,5,6) reverse_::: List(3,2,1)`

**Comment by lukaseder** [comment]
Thanks for the explanation. I see. I would not have guessted this 😁. Well, adding the parens again works well enough, too, so I won't add the space, for now

**Comment by joroKr21** [comment]
Hmm the parens might be a problem for people that have turned up their linters and compile with fatal warnings.

**Comment by lukaseder** [comment]
Which would be in conflict with https://github.com/jOOQ/jOOQ/issues/5422

Who would have thought that the Scala language is so sensitive with respect to what appears to be pure style to the untrained eye...

**Comment by lukaseder** [comment]
OK then, so let's fix this once more. Whenever we generate `def %s:` where `%s` is some identifier, we need to add a whitespace after the identifier if it ends with `[^a-zA-Z0-9]`

**Comment by lukaseder** [comment]
Well, I'm glad I reopened this. As the new commits show, there had been a lot of other cases where the `CLASS` attribute would have caused compilation issues, which hadn't been covered by integration tests.

**Comment by lukaseder** [comment]
Fixed again for 3.15.0 and new backport to 3.14.4 (#10878)

**Comment by joroKr21** [comment]
Awesome work 🥳
